Implement lab 04 numeronym

undrewb opened the pull request:
Implement functions
Implement tests
Fixes #517 

Review of colleague's PR #515

#### Changes proposed in this PR:

- Implement lab 4 of go-course
- Decompose into numeronym and numeronyms for testing.
- Don't consider non-ascii or special chars ( like _ or white space ) different to regular chars.
Implement testing.

Commit:
Implement lab 04 numeronym

Implement functions
Implement tests

sirigithub commented on 04_numeronym/undrewb/main.go:
```diff
@@ -0,0 +1,30 @@
+}
+
+func numeronym(s string) string {
+	outs := s
```
@undrewb , neat implementation.  You can possibly do without another variable outs here.
Just return the formatted string when you can build a numeronym or return the string s otherwise.

undrewb replied:
fixed

sirigithub commented on 04_numeronym/undrewb/main_test.go:
```diff
@@ -0,0 +1,72 @@
+		want:  []string{}},
+}
+
+func TestNumeronyms(t *testing.T) {
```
- How would the functions behave for test data involving unicode strings ?

undrewb replied:
excellent pickup! It didnt cope with unicode strings. Ive updated my implementation and added test cases.

Commit:
Implement unicode handling

Add test cases for unicode
Modify numeronym to use rune length not byte len

Add test cases for unicode

anzboi reviewed (approved):
LGTM. Great work

anzboi commented:
Apologies for taking so long to get to this.

juliaogris commented:
The go-course is now closed. Thank you very much for participating.
